File: 1. Study/0530 - 효율성(투포인터 알고리즘)/kimshin/4.연속부분수열2.py
def sol(m,arr):
    answer=0
    sum=0
    lt=0
    for rt in range(len(arr)):
        sum+=arr[rt]

        while(sum>m):
            lt+=1
            sum-=arr[lt]
            # if sum == m: answer+=1
        answer+=rt-lt+1
    return answer

a=[1,3,1,2,3]
print(sol(5,a))

# ??

File: 1. Study/0530 - 효율성(투포인터 알고리즘)/mojaeya/4_연속 부분수열 2.js
// let arr = [1, 3, 1, 2, 3];

// console.log(solution(5, arr));

// function solution(M, arr) {
//   let answer = 0;
//   let sum = 0;
//   let end = 0;
//   for (let st = 0; st < arr.length; st++) {
//     while (end < arr.length) {
//       sum += arr[end];
//       if (sum <= M) {
//         answer++;
//       } else {
//         end = st++;
//         sum = 0;
//         break;
//       }
//       end++;
//     }
//   }
//   return answer;
// }
